src/test/java/spoon/test/record/CtRecordTest.java

```diff
@@ -202,6 +207,15 @@ void testGenericTypeParametersArePrintedBeforeTheFunctionParameters() {
 		assertEquals("public record GenericRecord<T>(T a, T b) {}", record.toString());
 	}
 
+	@Test
+	void testBuildRecordModelWithStaticInitializer() {
+		// contract: a record can have static initializers
+		String code = "src/test/resources/records/WithStaticInitializer.java";
+		CtModel model = assertDoesNotThrow(() -> createModelFromPath(code));
+		List<CtAnonymousExecutable> execs = model.getElements(new TypeFilter<>(CtAnonymousExecutable.class));
+		assertThat(execs.size(), equalTo(2));
+	}
```

Comment:
Maybe use `ModelTest` here?

Reply:
As building the model failed, we can't use ModelTest here